app/src/main/java/com/fpvout/digiview/dvr/DVR.java

```diff
@@ -0,0 +1,166 @@
+package com.fpvout.digiview.dvr;
+
+import android.app.Activity;
+import android.media.MediaRecorder;
+import android.os.Environment;
+import android.os.Handler;
+import android.os.Looper;
+import android.util.Log;
+import android.widget.ImageButton;
+import android.widget.Toast;
+
+import com.fpvout.digiview.R;
+import com.fpvout.digiview.UsbMaskConnection;
+import com.fpvout.digiview.helpers.DataListener;
+import com.fpvout.digiview.helpers.Mp4Muxer;
+import com.fpvout.digiview.helpers.StreamDumper;
+import com.fpvout.digiview.helpers.ThreadPerTaskExecutor;
+
+import java.io.File;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+
+public class DVR {
+    private final Activity activity;
+    private boolean recordAmbientAudio;
+    private MediaRecorder recorder;
+    private boolean recording = false;
+    private static DVR instance;
+    private static final String DVR_LOG_TAG = "DVR";
+    private String defaultFolder = "";
+    private String ambietAudio;
+    private String videoFile;
+    private String dvrFile;
+    private String fileName;
+    private StreamDumper streamDumper;
+    private UsbMaskConnection connection;
+    private static Handler updateAfterRecord;
+    public static final String LATEST_THUMB_FILE = "latest.jpeg";
+
+    DVR(Activity activity, boolean recordAmbientAudio, Handler updateAfterRecord, UsbMaskConnection connection){
+        this.activity = activity;
+        this.connection = connection;
+        this.recordAmbientAudio = recordAmbientAudio;
+        this.updateAfterRecord = updateAfterRecord;
+        defaultFolder =  Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_PICTURES) + "/" + this.activity.getApplicationInfo().loadLabel(this.activity.getPackageManager()).toString();
+        streamDumper = new StreamDumper(activity, defaultFolder);
+    }
+
+    public static DVR getInstance(Activity context, boolean recordAmbientAudio, Handler updateAfterRecord, UsbMaskConnection connection){
```

Comment:
Keeping the context in a singleton might lead to a memory leak.

Reply:
i will change it . The thing is i wanted to make sure only one instance is used in the whole app.